Fix clang build warning in WebSocketTaskCurl

kshukuwa opened the pull request:
#### f10eb9211cd87039d8bcb1463885e87bb8a98b10
<pre>
Fix clang build warning in WebSocketTaskCurl
<a href="https://bugs.webkit.org/show_bug.cgi?id=246847">https://bugs.webkit.org/show_bug.cgi?id=246847</a>

Reviewed by Fujii Hironori.

A warning &apos;not handled in switch&apos; is detected in WebSocketTask::didReceiveData()
due to &apos;OpCodeContinuation&apos;, &apos;OpCodePong&apos; and &apos;OpCodeInvalid&apos;. To solve the warning,
add those enum definitions.

* Source/WebKit/NetworkProcess/curl/WebSocketTaskCurl.cpp:
(WebKit::WebSocketTask::didReceiveData):

Canonical link: <a href="https://commits.webkit.org/255904@main">https://commits.webkit.org/255904@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cf37457403346f3e2290dc9f0504c865a8ca8360

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93962 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3151 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/24527 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103596 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3167 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31390 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86284 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/99649 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99625 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2267 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80386 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/29285 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84193 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/83886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/72241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37779 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/17719 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35649 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18984 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4076 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39522 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41553 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41461 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38214 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebKit/NetworkProcess/curl/WebSocketTaskCurl.cpp:
```diff
@@ -203,6 +203,11 @@ void WebSocketTask::didReceiveData(WebCore::CurlStreamID, const WebCore::SharedB
+        case WebCore::WebSocketFrame::OpCodeContinuation:
+        case WebCore::WebSocketFrame::OpCodePong:
+        case WebCore::WebSocketFrame::OpCodeInvalid:
+            break;
```
I don't have a strong opinion here.  How about an idea of using `default` here?

kshukuwa replied:
If a new enumerator is added, we can get a new warning and can figure out if we want to handle that case or not.

fujii replied:
Sounds good.

kshukuwa committed:
Fix clang build warning in WebSocketTaskCurl
https://bugs.webkit.org/show_bug.cgi?id=246847

Reviewed by Fujii Hironori.

A warning 'not handled in switch' is detected in WebSocketTask::didReceiveData()
due to 'OpCodeContinuation', 'OpCodePong' and 'OpCodeInvalid'. To solve the warning,
add those enum definitions.

* Source/WebKit/NetworkProcess/curl/WebSocketTaskCurl.cpp:
(WebKit::WebSocketTask::didReceiveData):

Canonical link: https://commits.webkit.org/255904@main
